### PR TITLE
Finish delete user

### PR DESCRIPTION
**Pull request by manhnhhedspi**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:edit, :update]
-  before_action :load_user, only: [:edit, :update, :show]
+  before_action :logged_in_user, only: [:edit, :update, :destroy]
+  before_action :load_user, only: [:edit, :update, :show, :destroy]
```
Comment (yuyuvn) [inline app/controllers/users_controller.rb]: hơn 3 cái thì except cho ngắn

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -13,6 +14,7 @@ def new
 
   def create
     @user = User.new user_params
+    @user.role = Settings.role.normal_user
```
Comment (yuyuvn) [inline app/controllers/users_controller.rb]: dùng merge params

Comment (vuducdat) [inline app/controllers/users_controller.rb]: you can use enum.

File: config/locales/en.yml [hunk 3]
```diff
@@ -76,3 +77,8 @@ en:
     sessions:
       create:
         flash_danger: "Invalid email/password combination"
+  admin:
+   users:
+    user:
```
Comment (yuyuvn) [inline config/locales/en.yml]: indent

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -8,7 +8,11 @@
...
         <li><%= link_to t(".help"), "#" %></li>
         <% if logged_in? %>
-          <li><%= link_to t(".all_users"), users_path %></li>
+          <% if is_admin? %>
+            <li><%= link_to t(".all_users"), admin_users_path %></li>
+          <% else %>
+            <li><%= link_to t(".all_users"), users_path %></li>
+          <% end %>
           <li class="dropdown">
             <%= link_to "#", class: "dropdown-toggle", "data-toggle" => "dropdown" do %>
```
Comment (biznezo) [inline app/views/layouts/_header.html.erb]: try to remove =>

Comment (biznezo) [inline app/views/layouts/_header.html.erb]: data-toggle="dropdown"

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -19,6 +19,10 @@ def logged_in?
     !current_user.nil?
   end
 
+  def is_admin?
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: `admin?`

File: config/routes.rb [hunk 6]
```diff
@@ -7,4 +7,7 @@
   resources :users
   resources :relationships, only: [:create, :destroy]
   resources :categories, only: [:index]
+  namespace :admin do
+    resources :users, only: [:index, :destroy]
```
Comment (dieunb) [inline config/routes.rb]: root_path

File: app/controllers/admin/base_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class Admin::BaseController < ApplicationController
+  before_action :admin_role
```
Comment (dieunb) [inline app/controllers/admin/base_controller.rb]: `verify_admin`, how do you think?

File: app/models/user.rb [hunk 8]
```diff
@@ -1,6 +1,15 @@
 class User < ActiveRecord::Base
   attr_accessor :remember_token
+
+  enum role: {normal_user: 0,
```
Comment (dieunb) [inline app/models/user.rb]: simple
```
+  enum role: [:normal, :admin, :teacher, :blocked]
```

**Pull request merged**